File: src/java/Webapp/usr/fne/common/Interval.java
//----------------------------------------------------------------------------
//
//       Copyright (C) 2008 Frank Eskesen.
//
//       This file is free content, distributed under the GNU General
//       Public License, version 3.0.
//       (See accompanying file LICENSE.GPL-3.0 or the original
//       contained within https://www.gnu.org/licenses/gpl-3.0.en.html)
//
//----------------------------------------------------------------------------
//
// Title-
//       Interval.java
//
// Purpose-
//       Time interval calculator.
//
// Last change date-
//       2008/01/01
//
//----------------------------------------------------------------------------
package usr.fne.common;

import java.io.*;
import java.lang.*;
import java.util.*;

public class Interval
{
//----------------------------------------------------------------------------
// Interval.Attributes
//----------------------------------------------------------------------------
Date                   time1;       // The start time

//----------------------------------------------------------------------------
//
// Method-
//       Interval.Interval
//
// Purpose-
//       Constructor.
//
//----------------------------------------------------------------------------
public
   Interval( )                    // Constructor
{
   time1= new Date();
}

//----------------------------------------------------------------------------
//
// Method-
//       Interval.get
//
// Purpose-
//       Compute the Interval, in milliseconds, since the Interval was
//       constructed or the time function was called.
//
//----------------------------------------------------------------------------
public long                         // The interval, in milliseconds
   get( )                           // Compute interval
{
   long                result;      // Resultant

   Date time2= new Date();
   result= time2.getTime() - time1.getTime();
   time1= time2;

   return result;
}
} // Class Interval

File: src/cpp/lib/com/Interval.cpp
//----------------------------------------------------------------------------
//
//       Copyright (c) 2007-2014 Frank Eskesen.
//
//       This file is free content, distributed under the GNU General
//       Public License, version 3.0.
//       (See accompanying file LICENSE.GPL-3.0 or the original
//       contained within https://www.gnu.org/licenses/gpl-3.0.en.html)
//
//----------------------------------------------------------------------------
//
// Title-
//       Interval.cpp
//
// Purpose-
//       Interval Timer.
//
// Last change date-
//       2014/01/01
//
//----------------------------------------------------------------------------
#include "com/Interval.h"

//----------------------------------------------------------------------------
//
// Method-
//       Interval::~Interval
//
// Purpose-
//       Destructor.
//
//----------------------------------------------------------------------------
   Interval::~Interval( void )      // Destructor
{
}

//----------------------------------------------------------------------------
//
// Method-
//       Interval::Interval
//
// Purpose-
//       Constructor.
//
//----------------------------------------------------------------------------
   Interval::Interval( void )       // Constructor
:  startTime()
,  stopTime()
{
   start();
}

   Interval::Interval(              // Constructor
     const Clock&      start,       // Start time
     const Clock&      stop)        // Stop  time
:  startTime(start)
,  stopTime(stop)
{
}

//----------------------------------------------------------------------------
//
// Method-
//       Interval::operator double
//
// Purpose-
//       Cast to double
//
//----------------------------------------------------------------------------
   Interval::operator double( void ) const // Cast to double
{
   return stopTime.getTime() - startTime.getTime();
}

//----------------------------------------------------------------------------
//
// Method-
//       Interval::toDouble
//
// Purpose-
//       Cast to double.
//
//----------------------------------------------------------------------------
double                              // Time difference
   Interval::toDouble( void ) const // Cast to double
{
   return operator double();
}

//----------------------------------------------------------------------------
//
// Method-
//       Interval::start
//
// Purpose-
//       Start the interval timer.
//
//----------------------------------------------------------------------------
double                              // Current time difference
   Interval::start( void )          // Start the Interval Timer
{
   stopTime= startTime= Clock::current();
   return 0.0;
}

//----------------------------------------------------------------------------
//
// Method-
//       Interval::stop
//
// Purpose-
//       Stop the interval timer.
//
//----------------------------------------------------------------------------
double                              // Current time difference
   Interval::stop( void )           // Stop the interval timer
{
   stopTime= Clock::current();
   return operator double();
}

